### PR TITLE
Fix: Preview error viewer in platform and IE11

### PR DESCRIPTION
**Pull request by tonyjin**
- Contact Us redirect to Box support page should only show up in the Box web application
- Fix non-wrapping error text in IE11
- Consolidate download checks into util function
- Move file-specific util method into file.js

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @tonyjin has signed the CLA. Thanks for the pull request!

File: src/lib/util.js [hunk 1]
```diff
@@ -887,3 +875,12 @@ export function getProp(object, propPath, defaultValue) {
...
     return value !== undefined ? value : defaultValue;
 }
+
+/**
+ * Returns whether Preview is running inside the Box web application.
+ *
+ * @return {boolean} Is Preview running in the Box WebApp
+ */
+export function inBoxWebApp() {
+    return (window.location.hostname || '').indexOf('app.box.com') !== -1;
```
Comment (jeremypress) [inline src/lib/util.js]: We have an 'includes' polyfill if you want to use that
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes

File: src/lib/__tests__/util-test.js [hunk 2]
```diff
@@ -812,4 +786,12 @@ describe('lib/util', () => {
             expect(util.getProp(a, 'foo.bar', 'default')).to.equal('default');
         });
     });
+
+    describe('inBoxWebApp()', () => {
+        it('should return false when window location is not a Box domain', () => {
+            expect(util.inBoxWebApp()).to.be.false;
+        });
+
+        /* No quick way to stub window.location.hostname to test for truth case */
```
Comment (jeremypress) [inline src/lib/__tests__/util-test.js]: According to SO this works for href, maybe will work for hostname too?
```
global.window = {
       location: {
           href: {
               value: 'foo'
           }
       }
   }
```

Comment (tonyjin) [reply src/lib/__tests__/util-test.js]: I found that thread and wasted ~an hour yesterday trying to get the test to work unsuccessfully. Tried a few things including Object.defineProperty, global.window (which wouldn't work in the browser anyways I think), and a few others.

File: src/lib/util.js (hunk 1 above)
Comment (tonyjin) [reply src/lib/util.js]: Actually I'm gonna hold off adding a polyfill for now. The existing polyfill is for Array.prototype.includes. If I included the String polyfill, I'd want to update all references of `indexOf` and that looks like quite a few files.

Comment (DanDeMicco) [reply src/lib/util.js]: Seems like we follow an isSomething format for boolean returning functions. Maybe isBoxWebApp, or the inverse function isPlatformApp

File: src/lib/__tests__/Preview-test.js [hunk 3]
```diff
@@ -1721,64 +1690,39 @@ describe('lib/Preview', () => {
...
+        it('should show print button if print is supported', () => {
+            stubs.checkFeature.withArgs(sinon.match.any, 'print').returns(true);
             stubs.canDownload.returns(true);
-
             preview.finishLoading();
-            expect(stubs.showDownloadButton).to.be.called;
+            expect(stubs.showPrintButton).to.be.called;
+        });
 
+        it('should show not print button if print is not supported', () => {
```
Comment (DanDeMicco) [inline src/lib/__tests__/Preview-test.js]: should not show

File: src/lib/__tests__/Preview-test.js [hunk 4]
```diff
@@ -1721,64 +1690,39 @@ describe('lib/Preview', () => {
...
-
             preview.finishLoading();
-            expect(stubs.showDownloadButton).to.be.calledWith(preview.download);
+            expect(stubs.showPrintButton).to.not.be.called;
         });
 
-        it('should show print button if print is supported', () => {
-            stubs.checkFeature.returns(false);
-
+        it('should show not print button if file can\'t be downloaded', () => {
```
Comment (DanDeMicco) [inline src/lib/__tests__/Preview-test.js]: should not show

**Pull request merged**